### PR TITLE
Add initial i.MX RT 1020 support

### PR DESCRIPTION
**Pull request by Florob**
This adds basic support for i.MX RT 1020 chips.
It has been verified on the EVK as well as on a custom board design. Not in every possible module/pin configuration though.

Peripherals supported by this are:
* LPI2C
* LPSPI
* LPUART

### CODE REVIEW

**Comment by teburd** [comment]
Clippy failure should be fixed, otherwise this looks good to me

**Review by mciantyre — COMMENTED** [review]
LPSPI, LPUART, LPI2C, and CCM pins match what's in rev 2 of the 1020 reference manual. I just have one question about the newly-added default features.

I pushed a clippy fix on `master`. We'll rebase before merging.

The `lpspi::Pins` modification is a breaking change. But, it looks like we could rewrite 799d0fb07dd955b35b43a8f3e18e43590d0ef95b by dropping the optional LPSPI daisies and the LPSPI3 implementations. We could then release this within the 0.2 series, making nearly all of the work available to existing users. I plan to try this approach unless someone disagrees.

File: Cargo.toml [hunk 1]
```diff
@@ -13,8 +13,10 @@ keywords = ["imxrt", "nxp", "embedded", "no_std"]
 categories = ["embedded", "no-std"]
 
 [features]
+default = ["imxrt1010", "imxrt1020", "imxrt1060"]
```
[the remark this answers is not in the document]
Comment (Florob) [reply Cargo.toml]: That's an oversight on my part. I meant to leave that out of the commit. I had it for rust-analyzer/easier cargo check runs.

**Comment by Florob** [comment]
> The lpspi::Pins modification is a breaking change. But, it looks like we could rewrite https://github.com/imxrt-rs/imxrt-iomuxc/commit/799d0fb07dd955b35b43a8f3e18e43590d0ef95b by dropping the optional LPSPI daisies and the LPSPI3 implementations. We could then release this within the 0.2 series, making nearly all of the work available to existing users. I plan to try this approach unless someone disagrees.

It's a bit unfortunate for me personally, because LPSPI3 happens to be exactly the bus my own design is using. However, generally this sounds like a very sensible approach. I've taken the liberty to update the PR with changes to that effect.

**Comment by mciantyre** [comment]
Thanks for the update. 67a7bf6 applies cleanly into the 0.2 series. I'll release that within the next few days.

Just to clarify my previous comment: we will include the `lpspi::Pins` trait change and include support for your LPSPI3 implementation in the 0.3 release series. I pushed a re-write of 64374320ff473c18a3aa7928dfe5019bc9227c68 to include in this merge.

I have other minor, breaking changes that I'm considering throughout the imxrt-rs ecosystem. I expect I'll need a few weeks to play with ideas and make releases.

**Comment by Florob** [comment]
> Sorry, I wasn't clear in my previous comment. We will include the lpspi::Pins trait change and include support for your LPSPI3 implementation in the 0.3 release series.

Oh, I did understand that. I just was confused what state you wanted master in, since there is no separate 0.2 branch (yet).

49f1fa189a66284e11b8c22beab9b49c28bfda68 should probably also remove the TODO comment I've added to SPI3.

**Pull request merged**